haystack_experimental/components/converters/pdf_to_image.py

```diff
@@ -0,0 +1,139 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import mimetypes
+from pathlib import Path
+from typing import Any, Dict, List, Literal, Optional, Union
+
+from haystack import component, logging
+from haystack.components.converters.utils import get_bytestream_from_source, normalize_metadata
+from haystack.dataclasses import ByteStream
+from haystack.utils import expand_page_range
+
+from haystack_experimental.components.converters.image_utils import DETAIL_TO_IMAGE_SIZE, read_image_from_pdf
+from haystack_experimental.dataclasses.chat_message import ImageContent
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class PDFToImageContent:
+    """
+    Converts PDF files to ImageContent objects.
+    """
+
+    def __init__(
+        self,
+        *,
+        detail: Optional[Literal["auto", "high", "low"]] = None,
+        downsize: bool = False,
+        page_range: Optional[List[Union[str, int]]] = None,
+    ):
+        """
+        Create the PDFToImageContent component.
+
+        :param detail: Optional detail level of the image (only supported by OpenAI). One of "auto", "high", or "low".
+        :param downsize: If True, the image will be downscaled to the specified detail level.
+        :param page_range: A range of pages to extract metadata from. For example, page_range=['1', '3'] will extract
```

Comment:
If I understand well, `page_range` indicates the pages to convert to images.
E.g., if I have a PDF of 100 pages and I specify `page_range=['1', '3']`, I will get a list of 2 `ImageContent` objects.

If I am correct, we should fix the docstrings accordingly and communicate this aspect.

Reply:
I've updated the docstring. Let me know if it's more clear!